extensions/qxsck/var-and-list.js

```diff
@@ -5,292 +5,872 @@
+            text: Scratch.translate("sort list [LIST] with [CASE]"),
+            arguments: {
+              LIST: {
+                type: "string",
+                defaultValue: "list",
+              },
+              CASE: {
+                type: "string",
+                menu: "sortList.List",
+              },
+            },
+          },
+          {
+            opcode: "sortListRange",
+            blockType: "command",
+            text: Scratch.translate(
+              "sort item [LEFT] to [RIGHT] in list [LIST] with [CASE]"
+            ),
+            arguments: {
+              LIST: {
+                type: "string",
+                defaultValue: "list",
+              },
+              CASE: {
+                type: "string",
+                menu: "sortList.List",
+              },
+              LEFT: {
+                type: "string",
+                defaultValue: "1",
+              },
+              RIGHT: {
+                type: "string",
+                defaultValue: "2",
+              },
+            },
+          },
+          {
+            opcode: "isOrder",
+            blockType: "Boolean",
+            text: Scratch.translate("list [LIST] is [CASE] ?"),
+            arguments: {
+              LIST: {
+                type: "string",
+                defaultValue: "list",
+              },
+              CASE: {
+                type: "string",
+                menu: "isOrder.List",
+              },
+            },
+          },
+          {
+            opcode: "mapObject",
+            blockType: "command",
+            text: Scratch.translate(
+              "map object [OBJ] , keys to list [LIST], and values to list [LIST2]"
+            ),
+            arguments: {
+              OBJ: {
+                type: "string",
+                defaultValue: '{"name":"TruboWarp"}',
+              },
+              LIST: {
+                type: "string",
+                defaultValue: "list1",
+              },
+              LIST2: {
+                type: "string",
+                defaultValue: "list2",
+              },
+            },
+          },
+          {
+            opcode: "associateList",
+            blockType: "reporter",
+            text: Scratch.translate(
+              "associate list [LIST] to keys, and [LIST2] to values, if length is different, then [DO]"
+            ),
+            arguments: {
+              LIST: {
+                type: "string",
                 defaultValue: "list1",
               },
               LIST2: {
-                type: Scratch.ArgumentType.STRING,
+                type: "string",
                 defaultValue: "list2",
               },
+              DO: {
+                type: "string",
+                menu: "associateList.List",
+              },
+            },
+          },
+
+          {
+            opcode: "unSupportCompile",
+            blockType: "label",
+            text: Scratch.translate("⚠️Compilation not supported"),
```

Comment:
Compilation **is** supported, you're on TurboWarp, not Gandi IDE.

Reply:
does now Turbowarp supports compiling C-type blocks?

Reply:
yes

Reply:
> yes

ok,delete it just now